### PR TITLE
feat(v4-examples): migrate router provider examples starts with "c" and "b"

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
- base-antd
- base-chakra-ui
- base-headless
- base-mantine
- base-mui
- calendar-app
- command-palette-kbar
- core-use-import
- core-use-menu
- core-use-modal
- core-use-select
- customization-footer
- customization-login
- customization-offlayout-area
- customization-rtl
- customization-sider
- customization-theme-antd
- customization-theme-chakra-ui
- customization-theme-mantine
- customization-theme-mui
- customization-top-menu-layout

---------- 

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: examples/calendar-app/src/App.tsx [hunk 1]
```diff
@@ -10,17 +11,35 @@ const API_URL = "https://api.fake-rest.refine.dev";
...
-            Layout={Layout}
-        />
+        <BrowserRouter>
+            <Refine
+                routerProvider={routerProvider}
+                dataProvider={dataProvider(API_URL)}
+                resources={[
+                    {
+                        name: "events",
+                        list: CalendarPage,
```
Comment (salihozdemir) [inline examples/calendar-app/src/App.tsx:21]:
```suggestion
                        list: "/events",
```

File: examples/core-use-import/src/App.tsx [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
-            resources={[{ name: "posts", list: DummyList }]}
-        />
+        <BrowserRouter>
+            <Refine
+                dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
+                routerProvider={routerProvider}
+                resources={[{ name: "posts", list: "/posts" }]}
+            >
+                <Routes>
+                    <Route element={<Outlet />}>
```
Comment (salihozdemir) [inline examples/core-use-import/src/App.tsx:17]:
```suggestion                    
```
I think, there is no need to add parent `Route` in here.

File: examples/core-use-modal/src/App.tsx [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
-            resources={[{ name: "posts", list: DummyList }]}
-        />
+        <BrowserRouter>
+            <Refine
+                dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
+                routerProvider={routerProvider}
+                resources={[{ name: "posts", list: "/posts" }]}
+            >
+                <Routes>
+                    <Route element={<Outlet />}>
```
Comment (salihozdemir) [inline examples/core-use-modal/src/App.tsx:17]:
```suggestion
```

File: examples/core-use-select/src/App.tsx [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
-            resources={[{ name: "posts", list: DummyList }]}
-        />
+        <BrowserRouter>
+            <Refine
+                dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
+                routerProvider={routerProvider}
+                resources={[{ name: "posts", list: "/posts" }]}
+            >
+                <Routes>
+                    <Route element={<Outlet />}>
```
Comment (salihozdemir) [inline examples/core-use-select/src/App.tsx:17]:
```suggestion
```

**Review by salihozdemir — CHANGES_REQUESTED** [review]

**Review by salihozdemir — APPROVED** [review]

**Review by BatuhanW — APPROVED** [review]

**Pull request merged**